api/views.go

```diff
@@ -82,3 +83,94 @@ func (c *Views) List() ([]ViewListItem, error) {
 
 	return q.View, graphqlErr
 }
+
+type ViewConnectionInput struct {
+	RepositoryName graphql.String `json:"repositoryName"`
+	Filter graphql.String `json:"filter"`
+}
+
+func (c *Views) Create(name, description string, connections map[string]string) error {
+	var m struct {
+		CreateView struct {
+			Name string
+			Description string
+		} `graphql:"createView(name: $name, description: $description, connections: $connections)"`
+	}
+                                                                                                
+	var viewConnections []ViewConnectionInput
+	for k, v := range connections {
+		viewConnections = append(
+			viewConnections,
+			ViewConnectionInput{
+				RepositoryName: graphql.String(k),
+				Filter: graphql.String(v),
+			})
+	}
+
+	variables := map[string]interface{} {
+		"name": graphql.String(name),
+		"description": graphql.String(description),
+		"connections": viewConnections,
+	}
+
+	graphqlErr := c.client.Mutate(&m, variables)
+
+	if graphqlErr != nil {
+		// The graphql error message is vague if the repo already exists, so add a hint.
+		return fmt.Errorf("%+v. Does the view already exist?", graphqlErr)
+	}
+
+	return nil
+}
+
+func (c *Views) Delete(name, reason string) error {
+	var m struct {
+		DeleteSearchDomain struct {
+			ClientMutationId string
+		} `graphql:"deleteSearchDomain(name: $name, deleteMessage: $reason)"`
+	}
+	variables := map[string]interface{}{
+		"name":   graphql.String(name),
+		"reason": graphql.String(reason),
+	}
+
+	err := c.client.Mutate(&m, variables)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Views) UpdateConnections(name string, connections map[string]string) error {
+	var m struct {
+		View struct {
+			Name    string
+		} `graphql:"updateView(viewName: $viewName, connections: $connections)"`
```

Comment:
It is not possible to update the description?

Reply:
At least not from the updateView operation.

The repository updates the description via updateDescriptionForSearchDomain. It might also work from views. I will check.

Reply:
And it works. I will update the command and api accordingly. Good catch.